beacon_node/network/src/sync/block_sidecar_coupling.rs

```diff
@@ -78,6 +96,35 @@ impl<E: EthSpec> BlocksAndBlobsRequestInfo<E> {
     }
 
     pub fn is_finished(&self) -> bool {
-        self.is_blocks_stream_terminated && self.is_sidecars_stream_terminated
+        let blobs_requested = matches!(self.request_type, ByRangeRequestType::BlocksAndBlobs);
```

Comment:
I think we should use a complete match here in case we add a new by range request type